smrf/data/modifiers.py

```diff
@@ -0,0 +1,43 @@
+def get_scalar_value_key(config, stage):
```

Comment:
Personal preference: As a person who likes OOP - I would make this a class `Modifiers` and make the two methods `@classmethod`. Then you would import that class and your code becomes a bit more descriptive at the places you call those methods.

Reply:
@scotthavens yeah I bet there is a more descriptive term. What if instead of `stage` it was a boolean for `pre_distribution` since it is either pre or post distribution? 

@jomey great call. I'll make that change.

Reply:
With going the class route, this could also be less dynamic and simply have two methods there. One for the input and one for the output key. Both key names can then be set as constants on the class and also used below on line#35 or have that also be replaced as a class constant to hold scalar keys options.
My idea would look like this:
```python
class Modifiers:
    INPUT_SCALAR_KEY = "input_scalar_factor"
    OUTPUT_SCALAR_KEY = "output_scalar_factor"
    KEY_OPTIONS = [INTPUT_SCALAR_KEY, OUTPUT_SCALAR_KEY]

    # two methods for the key retrieval
```

Reply:
The modifiers can only be applied in to places for the input and output, right? Lets just keep it verbose to that, `pre_distribution` makes an assumption on where the factor will be applied.

Building of @jomey  class style, would there be explict methods for inputs and outputs? We are already going to have to call these anyways in specific spots in the input/output code. Something along the lines below where `config` could be passed on `__init__` of the class and you explicitly ask to scale the input/output using the config and keys from above. There would be similar code for the outputs as well. It makes for a bit more code but a lot more understandable as to when the factors are being applied.

```python
def get_input_value(self):
    return self.config.get(self.INPUT_SCALAR_KEY)

def scale_input(self, data):
    return data * self.config.get(self.INPUT_SCALAR_KEY, None)
````

Reply:
I'm a big fan of the separate methods now that it was brought up. I'm just about done with a refactor to the class that should make it more verbose and make it extensible for other scaling approaches later on. I'll be pushing later today once I make the integration test changes.